src/scripts/services/user-service.ts

```diff
@@ -109,34 +110,54 @@ export class OktaUserService {
   };
 
   // eslint-disable-next-line functional/functional-parameters
-  readonly listUsers = (): TE.TaskEither<string, readonly User[]> => {
+  readonly listUsers = (
```

Comment:
At this stage, I'd prefer two simpler, exported functions: `listUser` and `listUsersInGroup`. Those functions may use some of the same private functions to do their job.

Reply:
So have three functions: `listUsers: () => TE.TaskEither<string, readonly User[]>`, `listUsersInGroup: (maybeGroup: O.Option<Group>) => TE.TaskEither<string, readonly User[]>`, and `privateListUsers: (clientOrGroup:okta.Client | okta.Group) => TE.TaskEither<string, readonly User[]>`. And have the first two call the second one?

There would be a lot of copy/pasted code if there were two functions that did both, the only difference is that one requires the fold and the other does not.

Reply:
Currently working on refactoring the code to meet the review, ignore previous comment

Reply:
@dspasojevic Code has been refactored to meet review in most recent commit, though a more thorough look at how it was done would be helpful